app/jobs/two_factor_authentication_removal_job.rb

```diff
@@ -0,0 +1,33 @@
+# For users in the process of 2fa removal
+
+class TwoFactorAuthenticationRemovalJob < ApplicationJob
+  queue_as :low
+
+  def perform
+    two_factor_authentication_removals = TwoFactorAuthenticationRemoval.all
+    two_factor_authentication_removals.each do |two_factor_authentication_removal|
+      publisher = Publisher.find(two_factor_authentication_removal.publisher_id)
+      # First 2 weeks of the job, send reminder e-mails
+      if !two_factor_authentication_removal.two_factor_authentication_removal_time_completed?
+        MailerServices::TwoFactorAuthenticationRemovalReminderEmailer.new(publisher: publisher).perform
+
+      # 14th day of the job
+      elsif two_factor_authentication_removal.two_factor_authentication_removal_time_completed? && !two_factor_authentication_removal.removal_completed?
+        # Delete 2fa registrations, channels, and disconnect wallet
+        publisher.totp_registration.destroy if publisher.totp_registration.present?
+        publisher.u2f_registrations.destroy_all if !publisher.u2f_registrations.empty?
+        publisher.channels.destroy_all if !publisher.channels.empty?
```

Comment:
Users need to still be paid. Please don't destroy, but you can remove the `publisher_id` on the `channels` table.

Reply:
Agreed, the users still need to, and will be paid; @evq's insight here was really helpful to me https://bravesoftware.slack.com/archives/C8GJLTQHW/p1552073068041900

The browser will hold funds until the publisher re-verifies their channels, for that reason we don't want channels to appear in the publishers list until they re-verify.